src/resolvers/resolver.ts

```diff
@@ -1,95 +1,121 @@
-                const token = jwt.sign(
-                    { userId: user._id, role: user._doc?.role || 'user' },
-                    SECRET,
-                    { expiresIn: '2h' }
-                )
-                const data = {
-                    token: token,
-                    user: { ...user._doc, password: '', role: user?.role || 'user' },
-                }
-                return data
-            } else {
-                throw new Error('Invalid credential')
-            }
-        },
-        async createProfile(
-            _: any,
-            { user, lastName, firstName, role }: any,
+    async getAllProfiles(_: any, _args: any, context: { userId: any }) {
+      if (!context.userId) throw new Error('Unauthorized')
+      const profiles = await Profile.find({})
+      return profiles
+  },
+  },
 
-        ) {
-            if (!mongoose.isValidObjectId(user)) throw new Error('Invalid user id')
-            const profileExists = await Profile.findOne({ where: { user } })
-            if (profileExists) throw new Error('User already have a profile')
-            const userExists = await User.findOne({ where: { _id: user } })
-            if (!userExists) throw new Error('This user does not exists')
-            const user = await User.findOne({ email: email })
-            const newProfile = await Profile.create({
-                lastName,
-                firstName,
-                role,
-                user,
-            })
-            await userExists.update({ profile: newProfile._id })
-            return newProfile
-        },
-    },
-    User: {
-        async profile(parent: any) {
-            return await Profile.findOne({ where: { user: parent._id.toString() } })
-        },
+  Mutation: {
+    async createUser(
+      _: any,
+      { registerInput: { email, password, role } }: any
+  ) {
+      const userExists = await User.findOne({ email: email })
+      if (userExists) throw new Error('Email is taken')
+      const emailExpression = /^(([^<>()\[\]\\.,;:\s@“]+(\.[^<>()\[\]\\.,;:\s@“]+)*)|(“.+“))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/
+      const isValidEmail = emailExpression.test(String(email).toLowerCase())
+      if (!isValidEmail)
+          throw new Error('invalid email format')
+      if (password.length < 6)
+          throw new Error('password should be minimum 6 characters')
+      const hashedPassword = await bcrypt.hash(password, 10)
+      const newUser = await User.create({
+          role: role || 'user',
+          email: email,
+          password: hashedPassword,
+      })
+      const token = jwt.sign(
+          { userId: newUser._id, role: newUser?.role },
+          SECRET,
+          { expiresIn: '2h' }
+      )
+
+      return { token, user: newUser }
+  },
+
+    async loginUser(_: any, { loginInput: { email, password } }: any) {
+      const user = await User.findOne({ email: email })
```

Comment:
Is this edited by you?

Reply:
@ceelogre
No, but you can leave feedback. I can take a  lead on that.

Reply:
I just see this as a very large file. Would love to break it down into utils.